tools/tck-api/src/main/scala/org/opencypher/tools/tck/api/CypherTCK.scala

```diff
@@ -64,57 +64,43 @@ object CypherTCK {
     */
   def allTckScenarios: Seq[Scenario] = parseClasspathFeatures(featuresPath).flatMap(_.scenarios)
 
+  @deprecated("use allTckScenarios instead")
   def allTckScenariosFromFilesystem: Seq[Scenario] = {
-    parseFilesystemFeatures(new File(getClass.getResource(featuresPath).toURI)).flatMap(_.scenarios)
+    allTckScenarios
   }
 
   def parseClasspathFeatures(path: String): Seq[Feature] = {
-    val resource = getClass.getResource(path).toURI
-    val fs = FileSystems.newFileSystem(resource, new util.HashMap[String, String]) // Needed to support `Paths.get` below
+    parseFeatures(getClass.getResource(path).toURI)
+  }
+
+  def parseFeatures(resource: URI): Seq[Feature] = {
+    val fs =
+      if ("jar".equalsIgnoreCase(resource.getScheme)) {
+        Some(FileSystems.newFileSystem(resource, new util.HashMap[String, String]))
+      } else None
     val directoryPath: Path = Paths.get(resource)
     try {
-      val featureUrls = parseClasspathFeatureURLsRecursive(directoryPath)
-      featureUrls.map(parseClasspathFeature)
+      val featurePaths = Files.walk(directoryPath).filter(new Predicate[Path] {
+        override def test(t: Path): Boolean = Files.isRegularFile(t) && t.toString.endsWith(featureSuffix)
```

Comment:
Oh how we long for java lambda support in Scala...

Reply:
After writing that IntelliJ taught me that we can actually write this a bit more elegantly:

```
      val featurePaths = Files.walk(directoryPath).filter {
        (t: Path) => Files.isRegularFile(t) && t.toString.endsWith(featureSuffix)
      }
```

Reply:
Adopted.